src/android/io/cozy/plugins/listlibraryitems/ListLibraryItems.java

```diff
@@ -545,4 +570,47 @@ private File createThumbnail(String filePath, String contentType) throws IOExcep
             return thumb;
         }
     }
+
+    private boolean listMediaBuckets(CallbackContext callbackContext) {
+        try {
+            final HashSet<String> queryResults = new HashSet<>();
+            queryResults.addAll(listMediaBucketsByUri(MediaStore.Images.Media.INTERNAL_CONTENT_URI));
+            queryResults.addAll(listMediaBucketsByUri(MediaStore.Video.Media.INTERNAL_CONTENT_URI));
+            queryResults.addAll(listMediaBucketsByUri(MediaStore.Images.Media.EXTERNAL_CONTENT_URI));
+            queryResults.addAll(listMediaBucketsByUri(MediaStore.Video.Media.EXTERNAL_CONTENT_URI));
+
+            PluginResult pluginResult = new PluginResult(PluginResult.Status.OK, new JSONArray(queryResults));
+            callbackContext.sendPluginResult(pluginResult);
+            return true;
+
+        } catch (Exception e) {
+            e.printStackTrace();
+            callbackContext.error(e.getMessage());
+            return false;
+        }
+    }
+
+    private HashSet<String> listMediaBucketsByUri(Uri uri) throws JSONException {
+        String[] PROJECTION_BUCKET = {
+                MediaStore.Images.Media.BUCKET_ID,
+                MediaStore.Images.Media.BUCKET_DISPLAY_NAME
+        };
+        String BUCKET_GROUP_BY = "1) GROUP BY 1,(2";
```

Comment:
Can we add a ORDER_BY BUCKET_DISPLAY_NAME ASC ?

Reply:
Yes, it should be possible to add an `ORDER BY` clause, but if you want to display an ordered bucket list to the user, I think it would be better to sort by bucket name in the react component.

Reply:
Fair enough